backend/market/serializers.py

```diff
@@ -106,84 +102,100 @@ def contains_profanity(self, text):
 
     def destroy(self, instance):
-        # Check if the user is the subletter before allowing the delete
+        # Check if the user is the seller before allowing the delete
         if (
-            self.context["request"].user == instance.subletter
+            self.context["request"].user == instance.seller
             or self.context["request"].user.is_superuser
         ):
             instance.delete()
         else:
-            raise serializers.ValidationError("You do not have permission to delete this sublet.")
-
+            raise serializers.ValidationError("You do not have permission to delete this item.")
 
-class SubletSerializerRead(serializers.ModelSerializer):
-    amenities = serializers.PrimaryKeyRelatedField(
-        many=True, queryset=Amenity.objects.all(), required=False
-    )
-    images = SubletImageURLSerializer(many=True, required=False)
 
+# Read-only serializer for use when pulling all items/etc
+class ItemSerializerRead(serializers.ModelSerializer):
     class Meta:
-        model = Sublet
-        read_only_fields = ["id", "created_at", "subletter", "sublettees"]
+        model = Item
         fields = [
             "id",
-            "subletter",
-            "amenities",
+            "seller",
+            "tags",
+            "category",
+            "favorites",
             "title",
-            "address",
-            "beds",
-            "baths",
-            "description",
-            "external_link",
             "price",
             "negotiable",
-            "start_date",
-            "end_date",
-            "expires_at",
             "images",
         ]
+        read_only_fields = fields
 
 
-# simple sublet serializer for use when pulling all serializers/etc
-class SubletSerializerSimple(serializers.ModelSerializer):
-    amenities = serializers.PrimaryKeyRelatedField(
-        many=True, queryset=Amenity.objects.all(), required=False
-    )
-    images = SubletImageURLSerializer(many=True, required=False)
+class SubletSerializer(serializers.ModelSerializer):
+    item = ItemSerializer(required=True)
 
     class Meta:
         model = Sublet
-        fields = [
-            "id",
-            "subletter",
-            "amenities",
-            "title",
-            "address",
-            "beds",
-            "baths",
-            "price",
-            "negotiable",
-            "start_date",
-            "end_date",
-            "images",
-        ]
-        read_only_fields = ["id", "subletter"]
+        fields = ["id", "item", "address", "beds", "baths", "start_date", "end_date"]
+        read_only_fields = ["id"]
+
+    def create(self, validated_data):
+        item_serializer = ItemSerializer(data=validated_data.pop("item"), context=self.context)
+        item_serializer.is_valid(raise_exception=True)
+        validated_data["item"] = item_serializer.save()
+        instance = super().create(validated_data)
+        return instance
+
+    def update(self, instance, validated_data):
```

Comment:
I think we can use the item serializer for update just like you did above? 

Also if you find that we don't need this explicit permission check on the serializer, lets remove it

Reply:
Fixed